src/otp-input/otp-input.tsx

```diff
@@ -118,62 +114,96 @@ export const OtpInput = ({
 
     const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
         setOtpValues(new Array(numOfInput).fill(""));
-        setEnableCountDownTimer(true);
+
+        if (!isWithinCooldown()) {
+            setLastCtaTimestamp(new Date());
+            setCountDown(cooldownDuration);
+        }
+
         if (onClick) {
             onClick(event);
         }
     };
 
+    // =========================================================================
+    // HELPER FUNCTIONS
+    // =========================================================================
+    const validateUserInput = (value: string, length = 1) =>
+        !value ? false : RegExp(`^[0-9]{${length}}$`).test(value);
+
+    const isWithinCooldown = (): boolean => {
+        const currentTime = Date.now();
+        const coolDownInMilliseconds = cooldownDuration * 1000;
+
+        return (
+            currentTime < lastCtaTimestamp.valueOf() + coolDownInMilliseconds
+        );
+    };
+
+    const beginCountdown = () => {
+        let timer = cooldownDuration;
+        const interval = setInterval(() => {
+            timer--;
+            setCountDown(timer);
+            if (timer === 0) {
+                if (onCooldownEnd) onCooldownEnd();
+                clearInterval(interval);
+            }
+        }, 1000);
+    };
+
+    const generateId = (index: number, name: string, prefix?: string) => {
+        return prefix
+            ? `${prefix}-${name}-${index + 1}`
+            : `${name}-${index + 1}`;
+    };
+
     // =============================================================================
     // RENDER FUNCTIONS
     // =============================================================================
-    const validateUserInput = (value: string, length = 1) => {
-        if (!value) {
-            return false;
-        }
-        return RegExp(`^[0-9]{${length}}$`).test(value);
-    };
-
     const renderCTALabel = () => {
         return otherCtaProps.children
             ? `${otherCtaProps.children}`
```

Comment:
return children directly as it could be a jsx element

```suggestion
            ? otherCtaProps.children
```